src/database/statement.rs

```diff
@@ -43,6 +43,75 @@ impl Statement {
             db_backend,
         }
     }
+
+    /// ```
+    /// # #[cfg(feature = "mock")]
+    /// # use sea_orm::{error::*, tests_cfg::*, MockDatabase, Transaction, DbBackend};
+    /// #
+    /// # let db = MockDatabase::new(DbBackend::Postgres)
+    /// #     .append_query_results(vec![vec![
+    /// #         maplit::btreemap! {
+    /// #             "name" => Into::<Value>::into("Chocolate Forest"),
+    /// #             "num_of_cakes" => Into::<Value>::into(1),
+    /// #         },
+    /// #         maplit::btreemap! {
+    /// #             "name" => Into::<Value>::into("New York Cheese"),
+    /// #             "num_of_cakes" => Into::<Value>::into(1),
+    /// #         },
+    /// #     ]])
+    /// #     .into_connection();
+    /// #
+    /// use sea_orm::{entity::*, query::*, tests_cfg::cake, FromQueryResult};
+    ///
+    /// #[derive(Debug, PartialEq, FromQueryResult)]
+    /// struct SelectResult {
+    ///     name: String,
+    ///     num_of_cakes: i32,
+    /// }
+    ///
+    /// # let _: Result<(), DbErr> = smol::block_on(async {
+    /// #
+    /// let res: Vec<SelectResult> = Statement::from_sql_and_values(
+    ///     DbBackend::Postgres,
+    ///     r#"SELECT "cake"."name", count("cake"."id") AS "num_of_cakes" FROM "cake""#,
+    ///     vec![],
+    /// )
+    /// .into_model::<SelectResult>()
+    /// .all(&db)
+    /// .await?;
+    ///
+    /// assert_eq!(
+    ///     res,
+    ///     vec![
+    ///         SelectResult {
+    ///             name: "Chocolate Forest".to_owned(),
+    ///             num_of_cakes: 1,
+    ///         },
+    ///         SelectResult {
+    ///             name: "New York Cheese".to_owned(),
+    ///             num_of_cakes: 1,
+    ///         },
+    ///     ]
+    /// );
+    /// #
+    /// # Ok(())
+    /// # });
+    ///
+    /// assert_eq!(
+    ///     db.into_transaction_log(),
+    ///     vec![Transaction::from_sql_and_values(
+    ///         DbBackend::Postgres,
+    ///         r#"SELECT "cake"."name", count("cake"."id") AS "num_of_cakes" FROM "cake""#,
+    ///         vec![]
+    ///     ),]
+    /// );
+    /// ```
+    pub fn into_model<M>(self) -> SelectorRaw<SelectModel<M>>
```

Comment:
I think we should rename `into_model()` to `select()`. As later on we might want `update()`, `delete()`...etc.

Reply:
Hmm, what's the difference between `select()` and  `update()`, `delete()` .. etc ?

Reply:
- Calling `select()` we get any type that impl `FromQueryResult`

- Calling `update()` we get `UpdateResult`

- Calling `delete()` we get `DeleteResult`

This just allow us to further extends the API of `Statement` further in the future.
Btw... we are not going to implement `update()` / `delete()` in this PR unless you want to loll.

Reply:
hmm, I named it `into_model()` to be consistent with the rest of the SeaORM API. If that's not good pick, we may remove this method from `Statement` and put it somewhere else ? 

Here are some options I can think of (I would prefer the 1st option if we are going to change this):
1) add a function to any type that impl `FromQueryResult` that would return `SelectorRaw<..>`
```rust
pub trait FromQueryResultExt: FromQueryResult + Sized {
    fn query_by(stmt: Statement) -> SelectorRaw<SelectModel<Self>> {
        SelectorRaw::<SelectModel<Self>>::from_statement(stmt)
    }
}

impl<T: FromQueryResult> FromQueryResultExt for T {}
```
This way we can use it like:
```rust
let res: Vec<SelectResult> = SelectResult::query_by(Statement::from_sql_and_values(
    DbBackend::Postgres,
    r#"SELECT "cake"."name", count("cake"."id") AS "num_of_cakes" FROM "cake""#,
    vec![],
))
.all(&db)
.await?;
```

2) maybe we should change `SelectorRaw::from_statement` to be public, and the example in the main comment would be:
```rust
let stmt = Statement::from_sql_and_values(
    DbBackend::Postgres,
    r#"SELECT "cake"."name", count("cake"."id") AS "num_of_cakes" FROM "cake""#,
    vec![],
);

let res: Vec<SelectResult> = SelectorRaw::from_statemnet(stmt)
.all(&db)
.await?;
```

Reply:
(2.) seems good to me.

Any comments? @tyt2y3

Reply:
maybe I should clarify why I preferred the 1st option, again consistency, option 1 follow the same way how entities get fetched:
```rust
// fetching entities 
let cheese: Option<cake::Model> = Cake::find(1).one(db).await?;

// fetching types that impl `FromQueryResult` using option 1
let res: Option<SelectResult> = SelectResult::query_by(stmt).one(db).await?;
```

Reply:
Actually... why not both? Both looks good.
Yes, and making a new Trait is a good idea too.
My only comment is perhaps `query` as the method name would suffice.

Reply:
> why not both? Both looks good.
just to make sure, here is the changes I'll made:
- remove `into_model()` from `Statemenet`
- write both options 1 and 2.

 >  perhaps query as the method name would suffice.
 I don't have strong preference here, if you like to see other options too:
 - `query()`
 - `find()` consistent with `Enitiy::find()`, I prefer this other the next one if are not going to add more variant of `find_*` in the future for `FromQueryResult`.
 - `find_by_statement()` consistent with `Entity::find_by_id()` and the name tells that it need `Statement` to be called

Reply:
I think this conversation should be resolved by https://github.com/SeaQL/sea-orm/pull/167/commits/e71a74b0c1cf4bc391733d35a1e8c2ac12269a04